File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> board)
{
    int r, c;
    int row = board.size();
    int col = board[0].size();
    
    int maxVal = 1;
    if(row>=2 && col >=2) {
        maxVal = 0;
    }
    for(r=1; r<row; ++r) {
        for(c=1; c<col; ++c) {
            if(board[r][c]==0) {
                continue;
            }
            board[r][c] = min({board[r][c-1], board[r-1][c], board[r-1][c-1]}) + 1;
            if(maxVal<board[r][c]) {
                maxVal = board[r][c];
            }
        }
    }
    maxVal *= maxVal;
    return maxVal;
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    maxVal = 0
    for x in range(len(board)):
        for y in range(len(board[0])):
            if board[x][y] == 1:
                maxVal = 1
    
    for x in range(1, len(board)):
        for y in range(1, len(board[0])):
            if board[x][y] == 0:
                continue
            board[x][y] = min(board[x-1][y], board[x][y-1], board[x-1][y-1]) + 1
            if maxVal < board[x][y]:
                maxVal = board[x][y]
    return (maxVal * maxVal)